src/gradualizer_bin.erl

```diff
@@ -65,11 +72,13 @@ bin_element_view({bin_element, _Anno, _Expr, default, Specifiers}) ->
         utf16     -> {0, 16}; %% 2-4 bytes
         utf32     -> {32, 0}  %% 4 bytes, fixed
     end;
-bin_element_view({bin_element, _Anno, _Expr, SizeSpec, Specifiers}) ->
+bin_element_view({bin_element, Anno, _Expr, SizeSpec, Specifiers} = BinElem) ->
     %% Non-default size, possibly a constant expression
-    try erl_eval:expr(SizeSpec, []) of
-        {value, Sz, _VarBinds} ->
-            {Sz * get_unit(Specifiers), 0}
+    try
+        case erl_eval:expr(SizeSpec, []) of
+            {value, Sz, _VarBinds} when is_integer(Sz) ->
+                {Sz * get_unit(Specifiers), 0}
+        end
```

Comment:
The exception is in `erl_eval:expr` right? If yes, please revert this unnecessary change. Then I think it's ready to merge.

Reply:
The expression might be in `erl_eval:expr` or a `case_clause`.

This change is intentional, please see the commit message for https://github.com/josefs/Gradualizer/pull/363/commits/8db6300fe62dcf5865b0163c8731715c0ccb3f60. From the official docs:

> In construction, `Size` is any expression that evaluates to an integer.

The point is that the `SizeSpec` must evaluate to int, so we use this syntax to ensure that `is_integer(Sz)` holds - if not, we handle the `case_clause`, too.

Reply:
Ah, ok, makes sense.

The case with only one clause can be written as `=` then. Not really important though.

    try
        {value, Sz, _VarBinds} = erl_eval:expr(SizeSpec, []),
        true = is_integer(Sz),
        {Sz * get_unit(Specifiers), 0}